taichi/program/program.cpp

```diff
@@ -177,12 +175,9 @@ FunctionType Program::compile(Kernel &kernel, OffloadedStmt *offloaded) {
   TI_AUTO_PROF;
   FunctionType ret = nullptr;
   if (arch_uses_llvm(config.arch) || kernel.arch == Arch::metal ||
```

Comment:
TODO: can we create a function to do this check in another PR? Right now it's repeated 3 times

Reply:
True, also will change some other nits.